homeassistant/components/media_player/kodi.py

```diff
@@ -15,20 +19,49 @@
     SUPPORT_TURN_OFF, MediaPlayerDevice, PLATFORM_SCHEMA)
 from homeassistant.const import (
     STATE_IDLE, STATE_OFF, STATE_PAUSED, STATE_PLAYING, CONF_HOST, CONF_NAME,
-    CONF_PORT, CONF_USERNAME, CONF_PASSWORD)
+    CONF_PORT, CONF_USERNAME, CONF_PASSWORD, EVENT_HOMEASSISTANT_STOP)
 import homeassistant.helpers.config_validation as cv
 
-REQUIREMENTS = ['jsonrpc-requests==0.3']
+REQUIREMENTS = ['websockets==3.2']
```

Comment:
It's possible, but I think even if we had a Kodi library, hass would still have to indicate which events it wanted to subscribe to. I'm not sure if there's a good way around it. I think that similar to the existing code, once we have the json-rpc abstracted and we're just calling object methods, it's going to look about the same as a dedicated Kodi library.